include/swift/AST/Types.h

```diff
@@ -2899,17 +2901,28 @@ class AnyFunctionType : public TypeBase {
     ExtInfo(Representation Rep,
             bool IsNoEscape,
             // SWIFT_ENABLE_TENSORFLOW
-            bool Throws, bool IsDifferentiable)
+            bool Throws, DifferentiabilityKind diffKind)
       : ExtInfo(Rep, Throws) {
       Bits |= (IsNoEscape ? NoEscapeMask : 0);
       // SWIFT_ENABLE_TENSORFLOW
-      Bits |= (IsDifferentiable ? DifferentiableMask : 0);
+      Bits |=
+          ((unsigned)diffKind << DifferentiableMaskOffset) & DifferentiableMask;
     }
 
     bool isNoEscape() const { return Bits & NoEscapeMask; }
     bool throws() const { return Bits & ThrowsMask; }
     // SWIFT_ENABLE_TENSORFLOW
-    bool isDifferentiable() const { return Bits & DifferentiableMask; }
+    bool isDifferentiable() const {
+      return ((DifferentiabilityKind)DifferentiableMask ==
+              DifferentiabilityKind::Normal) || isLinear();
+    }
+    bool isLinear() const {
```

Comment:
This method isn't necessary. All we need is a `getDifferentiabilityKind()` and `isDifferentiable()` (implemented in terms of `getDifferentiabilityKind()`).